examples/rt685s-evk/src/bin/flexspi-storage-service.rs

```diff
@@ -0,0 +1,524 @@
+    fn kind(&self) -> NorFlashErrorKind {
+        match self {
+            NorErrorType::FlashStorageErrorOther => NorFlashErrorKind::Other,
+            NorErrorType::FlashStorageErrorOutOfBounds => NorFlashErrorKind::OutOfBounds,
+            NorErrorType::FlashStorageErrorNotAligned => NorFlashErrorKind::NotAligned,
+        }
+    }
+}
+
+impl<T: BlockingNorStorageBusDriver> MacronixDeviceDriver<T, Blocking> {
+    pub fn get_jedec_id(&mut self, jedec: &mut [u8]) {
+        let read_cread_jedec_id_cmd = NorStorageCmd {
+            cmd_lb: 0x9F,
+            cmd_ub: Some(0x60),
+            addr: Some(0x0),
+            addr_width: Some(32),
+            bus_width: NorStorageBusWidth::Octal, // 3 - Octal
+            mode: NorStorageCmdMode::DDR,
+            dummy: NorStorageDummyCycles::Clocks(0x18),
+            cmdtype: Some(NorStorageCmdType::Read),
+            data_bytes: Some(4),
+        };
+
+        let _ = self
+            .storagebusdriver
+            .send_command(read_cread_jedec_id_cmd, Some(jedec), None);
+    }
+}
+
+impl<T: BlockingNorStorageBusDriver> BlockingReadNorFlash for MacronixDeviceDriver<T, Blocking> {
+    const READ_SIZE: usize = 1;
+
+    fn read(&mut self, offset: u32, bytes: &mut [u8]) -> Result<(), Self::Error> {
+        #[allow(const_evaluatable_unchecked)]
+        let mut read_start_ptr = 0;
+
+        if bytes.len() == 0 {
+            return Err(NorErrorType::FlashStorageErrorOther);
+        }
+
+        if bytes.len() % Self::READ_SIZE != 0 {
+            return Err(NorErrorType::FlashStorageErrorNotAligned);
+        }
+
+        if offset > self.capacity as u32 {
+            return Err(NorErrorType::FlashStorageErrorOutOfBounds);
+        }
+        if offset + bytes.len() as u32 > self.capacity as u32 {
+            return Err(NorErrorType::FlashStorageErrorOutOfBounds);
+        }
+
+        while read_start_ptr < bytes.len() {
+            // Read data from the storage device
+            let read_cmd = NorStorageCmd {
+                cmd_lb: 0xEE,
+                cmd_ub: Some(0x11),
+                addr: Some(offset + read_start_ptr as u32),
+                addr_width: Some(0x20),
+                bus_width: NorStorageBusWidth::Octal,
+                mode: NorStorageCmdMode::DDR,
+                dummy: NorStorageDummyCycles::Clocks(0x29),
+                cmdtype: Some(NorStorageCmdType::Read),
+                data_bytes: Some(Self::READ_SIZE as u32),
+            };
+
+            let _ = self.storagebusdriver.send_command(
+                read_cmd,
+                Some(&mut bytes[read_start_ptr..read_start_ptr + Self::READ_SIZE]),
+                None,
+            );
+
+            read_start_ptr += Self::READ_SIZE;
+        }
+
+        Ok(())
+    }
+
+    fn capacity(&self) -> usize {
+        self.capacity
+    }
+}
+
+impl<T: BlockingNorStorageBusDriver> BlockingNorFlash for MacronixDeviceDriver<T, Blocking> {
+    const WRITE_SIZE: usize = 256;
```

Comment:
This is a full page, but the flash also supports writing single bytes.
So this value should be 1